src/scss/tools/mixins/_generate_button_types.scss

```diff
@@ -23,15 +37,26 @@
     }
 
     @if map-has-key($buttonTypeConfig, border) {
-      border: solid 1px get-color(map-get($buttonTypeConfig, border));
+      $style: map-deep-get($buttonTypeConfig, border, style);
+      $color: map-deep-get($buttonTypeConfig, border, color);
+      border: get-border($style) get-color($color);
     }
 
     @if map-has-key($buttonTypeConfig, textColor) {
       color: get-color(map-get($buttonTypeConfig, textColor));
     }
 
+    @if map-has-key($buttonTypeConfig, opacity) {
+      opacity: map-get($buttonTypeConfig, opacity);
+    }
+
+    @if map-has-key($buttonTypeConfig, borderTopColor) {
+      border-top-color: get-color(map-get($buttonTypeConfig, borderTopColor));
+    }
+
     @if map-has-key($buttonTypeConfig, shadow) {
-      box-shadow: 6px 6px 20px 0 get-color(map-get($buttonTypeConfig, shadow));
+      $shadow: map-deep-get($buttonTypeConfig, shadow, kind);
+      box-shadow: map-get($shadows, $shadow) get-color((color: blackNeutral, key: darker, opacity: .2));
```

Comment:
@iondrimba did you try `get-shadow($shadow)` here or do we have some troubles with functions and variables with string as a key?